### PR TITLE
fix(components/breadcrumb): e2e

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this PR is trying to solve?**
Some items are hidden

**What is the chosen solution to this problem?**
Be ready to read all items for screen reader and add e2e tests

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Breadcrumbs/Breadcrumbs.component.js [hunk 1]
```diff
@@ -42,18 +42,31 @@ function Breadcrumbs(props) {
...
+				<Button id={id} bsStyle="link" role="link" title={title} onClick={wrappedOnClick}>
+					{text}
+				</Button>
+			) : (
+				<span id={id} title={title}>
+					{text}
+				</span>
+			);
+		}
+
```
Comment (jmfrancois) [inline packages/components/src/Breadcrumbs/Breadcrumbs.component.js]: I prefer to define a variable here. you have the if onclick just after and a lint error on wrappedOnClicl

```javascript
                let wrappedOnClick;
                let itemContent = <span id={id} title={title}>{text}</span>;
 		if (onClick) {
 			wrappedOnClick = event => onClick(event, item);
                         itemContent = <Button...
 		}
```

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    static final String COLLAPSED_MENU_SELECTOR = ".dropdown-menu";
+
+    static final String COLLAPSED_MENU_ITEM_SELECTOR = COLLAPSED_MENU_SELECTOR + " a";
+
+    /**
+     * Breadcrumb constructor
+     *
+     * @param driver Selenium WebDriver
+     */
+    Breadcrumb(WebDriver driver) {
```
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java]: You should add it in the `Application.java fields`. 
It is package private, so selenium projects classes won't be able to create it.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import org.openqa.selenium.WebElement;
+
+import java.util.Iterator;
+import java.util.List;
+
+public class Breadcrumb extends Component {
+
+    private static final Logger LOGGER = LogManager.getLogger(Breadcrumb.class);
+
+    static final String NAME = "Breadcrumb";
```
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java]: Make all those constants private please

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    Breadcrumb(WebDriver driver) {
+        super(driver, NAME, SELECTOR);
+    }
+
+    public List<WebElement> getItems() {
+        LOGGER.info(NAME + ".getItems");
+        return this.getElement().findElements(By.cssSelector(ITEM_SELECTOR));
+    }
+
+    public WebElement getItem(String label) {
```
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java]: IMO we should expose a `public void clickOnItem(final String label)`. In practice QA wants to get the item to click on it.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        LOGGER.info(NAME + ".getItem " + label);
+        Iterator<WebElement> elements = this.getItems().iterator();
+        while (elements.hasNext()) {
+            WebElement el = elements.next();
```
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java]:
```java
     for (final WebElement item : this.getItems()) {
            if (item.getText().equals(label)) {
                return item;
            }
        }
```

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    public List<WebElement> getCollapsedMenuItems() {
+        LOGGER.info(NAME + ".getCollapsedMenuItems");
+        return this.getElement().findElements(By.cssSelector(COLLAPSED_MENU_ITEM_SELECTOR));
+    }
+
+    public WebElement getCollapsedMenuItem(String label) {
+        LOGGER.info(NAME + ".getCollapsedMenuItem " + label);
+        Iterator<WebElement> elements = this.getCollapsedMenuItems().iterator();
+        while (elements.hasNext()) {
+            WebElement el = elements.next();
```
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java]:
```java
        for (WebElement menuItem : this.getCollapsedMenuItems()) {
            if (menuItem.getText().equals(label)) {
                return menuItem;
            }
        }
```

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        LOGGER.info(NAME + ".getCollapsedMenu");
+        return this.getElement().findElement(By.cssSelector(COLLAPSED_MENU_SELECTOR));
+    }
+
+    public List<WebElement> getCollapsedMenuItems() {
+        LOGGER.info(NAME + ".getCollapsedMenuItems");
+        return this.getElement().findElements(By.cssSelector(COLLAPSED_MENU_ITEM_SELECTOR));
+    }
+
+    public WebElement getCollapsedMenuItem(String label) {
```
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java]: Same here, this is usefull only to click on it, we should expose a `public void clickOnCollapsedMenuItem(final String label)`.
* click on menu button
* get the item with the label
* click on it

File: packages/components/e2e/component-objects/src/test/java/org/talend/component/JUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,3 @@
+package org.talend.component;
+
+interface JUnitTest {}
```
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/test/java/org/talend/component/JUnitTest.java]: Remove this please

File: packages/components/e2e/component-objects/src/test/java/org/talend/component/StorybookTest.java [hunk 9]
```diff
@@ -1,57 +1,35 @@
...
 
     protected void goToStory(final String categoryName, final String storyName) {
         this.goToMainElement();
 
-        final WebElement categoryMenu = driver.findElement(By.cssSelector(String.format(STORY_CATEGORY_SELECTOR, categoryName)));
-        final WebElement menusContainer = categoryMenu.findElement(By.xpath("following-sibling::*"));
-        if (menusContainer.findElements(By.tagName("ul")).size() == 0) {
-            categoryMenu.click();
+        try {
+            driver.get("http://localhost:6006/?selectedKind=" + categoryName + "&selectedStory=" + URLEncoder.encode(storyName, "UTF-8") + "&full=0&addons=1&stories=0&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel");
```
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/test/java/org/talend/component/StorybookTest.java]: Nice !

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+    private static final String SELECTOR = ".tc-breadcrumb";
+
+    private static final String ITEM_SELECTOR = ".tc-breadcrumb-item";
+
+    private static final String ACTIVE_ITEM_SELECTOR = ITEM_SELECTOR + ".active";
+
+    private static final String COLLAPSED_MENU_BUTTON_SELECTOR = ".dropdown-toggle";
+
+    private static final String COLLAPSED_MENU_SELECTOR = ".dropdown-menu";
```
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java]: COLLAPSED_MENU_BUTTON_SELECTOR : `SELECTOR + " .dropdown-toggle"`
COLLAPSED_MENU_SELECTOR : `SELECTOR  + " .dropdown-menu"`

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+     * Breadcrumb constructor
+     *
+     * @param driver Selenium WebDriver
+     */
+    public Breadcrumb(WebDriver driver) {
+        super(driver, NAME, SELECTOR);
+    }
+
+    public List<WebElement> getItems() {
+        LOGGER.debug(NAME + ".getItems");
```
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Breadcrumb.java]: What is the purpose of all those logs ?

File: packages/components/src/Breadcrumbs/Breadcrumbs.component.js [hunk 12]
```diff
@@ -63,30 +64,38 @@ function Breadcrumbs(props) {
...
+		function getItemContent() {
+			return !isActive && onClick ? (
+				<Button id={id} bsStyle="link" role="link" title={title} onClick={wrappedOnClick}>
+					{text}
+				</Button>
+			) : (
+				<span id={id} title={title}>
+					{text}
+				</span>
+			);
```
Comment (talend-glorieux) [inline packages/components/src/Breadcrumbs/Breadcrumbs.component.js]: Refrain from using multi-line ternaries.
Use regular `if/else` statements.

Comment (ncomont) [reply packages/components/src/Breadcrumbs/Breadcrumbs.component.js]: :-1:

**Pull request merged**